DragaliaAPI/Services/SavefileWriteService.cs

```diff
@@ -1,70 +1,155 @@
-﻿using DragaliaAPI.Models.Data;
+﻿using System;
+using System.Collections.Immutable;
+using DragaliaAPI.Models.Data;
+using DragaliaAPI.Models.Data.Entity;
+using DragaliaAPI.Models.Database;
 using DragaliaAPI.Models.Database.Savefile;
 using DragaliaAPI.Models.Dragalia.Responses.Common;
 using DragaliaAPI.Models.Dragalia.Responses.UpdateData;
+using DragaliaAPI.Services.Data;
 using Microsoft.EntityFrameworkCore;
 
 namespace DragaliaAPI.Services;
 
 public class SavefileWriteService : ISavefileWriteService
 {
     private readonly IApiRepository apiRepository;
+    private readonly ApiContext apiContext;
+    private readonly IUnitDataService unitDataService;
 
-    public SavefileWriteService(IApiRepository apiRepository)
+    public SavefileWriteService(
+        IApiRepository apiRepository,
+        ApiContext apiContext,
+        IUnitDataService unitDataService
+    )
     {
         this.apiRepository = apiRepository;
+        this.apiContext = apiContext;
+        this.unitDataService = unitDataService;
     }
 
     public async Task<UpdateDataList> CommitSummonResult(
-        List<SummonEntity> summonResult,
+        List<SimpleSummonReward> summonResult,
         string deviceAccountId,
         bool giveDew = true
     )
     {
         int dewGained = 0;
```

Comment:
not adding eldwater to savefile?

Reply:
probably a remnant of removing it from the commit and moving it to the summoncontroller method